File: Python/Credito.py
edad_cliente = int(input("Ingrese la edad del cliente: "))
ingresos = float(input("\nIngrese los ingresos mensuales del cliente: "))
fiador = input("\n¿El cliente tiene un fiador? (sí/no): ").lower()
reportado = input("\n¿El cliente ha sido reportado por incumplimiento de pagos? (sí/no): ").lower() == "sí"

if fiador == "sí":
    fiador = True

elif fiador == "no":
    fiador = False

else:
    print("Entrada no válida para el fiador. Por favor, ingrese 'sí' o 'no'.")

edad = (edad_cliente >= 18 and edad_cliente <= 75)
ingresos_suficientes = (ingresos >= 2000000 or fiador)
no_reportado = not reportado
credito_aprobado = edad and ingresos_suficientes and no_reportado

mensaje = "Crédito aprobado" if credito_aprobado else "Crédito denegado"

if credito_aprobado:
    print("\n¡Crédito aprobado!")
else:
    print("\nCrédito denegado.")

print("\n¿Edad del cliente es válida?", edad)
print("\n¿Ingresos del cliente son suficientes?", ingresos_suficientes)
print("\n¿El cliente no ha sido reportado?", no_reportado)
print("\n¿Crédito aprobado?", credito_aprobado, mensaje)

File: aJava/src/Credito.java
void main() {
    try (Scanner scanner = new Scanner(System.in)) {
        IO.println("Ingrese su edad: ");
        int edad = scanner.nextInt();

        IO.println("\nIngrese su salario mensual: ");
        double ingresos = scanner.nextDouble();

        IO.println("\n¿Tiene un fiador? (1. Sí, 2. No): ");
        boolean fiador = scanner.nextInt() == 1;

        IO.println("\n¿Esta reportado? (1. Sí, 2. No): ");
        boolean reportado = scanner.nextInt() == 1;

        boolean edadValida = (edad >= 18 && edad <= 75);
        boolean ingresosValidos = (ingresos >= 2000000) || fiador;
        boolean historialLimpio = !reportado;
        boolean creditoAprobado = edadValida && ingresosValidos && historialLimpio;

        IO.println("\nResultado: ");
        IO.println("\nEdad válida: " + (edadValida ? "Sí" : "No"));
        IO.println("Ingresos válidos: " + (ingresosValidos ? "Sí" : "No"));
        IO.println("Historial limpio: " + (historialLimpio ? "Sí" : "No"));
        IO.println("Crédito aprobado: " + (creditoAprobado ? "Sí" : "No"));
    }
}